### ISSUE TITLE
Serialized names from listing states

### ISSUE THREAD

**Issue by ClementN** [spatie/laravel-model-states#96]
Hello,

Is it possible to get the serialized names of the state when using

getStates(),  getStatesFor('state') and getDefaultStateFor('state')  ?

Thank you very much,

**Comment by pvanhemmen** [comment]
I have extended my abstract state class with a getter:

```
abstract class PositionState extends State
{
    /**
     * Get the serialized state names
     *
     * @return Collection
     * @throws ReflectionException
     */
    public static function getSerializedNames(): Collection
    {
        $names = collect([]);

        foreach (self::all() as $stateClass) {
            $reflection = new ReflectionClass($stateClass);
            $names->add($reflection->getStaticPropertyValue('name'));
        }

        return $names;
    }
}
```

Whenever needed it's now possible to obtain all serialized names with `PositionState::getSerializedNames()`. Would be great to have something like this baked into the State class itself, because it's quite useful for generating a list of filterable attributes for example. Would be willing to create a pull request if anyone's interested.

**Comment by pvanhemmen** [comment]
The necessary cange was jsut implemented in v2.0.0: https://github.com/spatie/laravel-model-states/blob/master/CHANGELOG.md
`HasStates::getStates()` and `HasStates::getStatesFor($field)` does exactly what's needed.

**Comment by brendt** [comment]
That's right, v2 should solve this problem.
